### ISSUE TITLE
ImageNet: spikes in training loss once per epoch

### ISSUE THREAD

**Issue by yaroslavvb** [diux-dev/cluster#43]
Seems like a bug in cross entropy calculation due to uneven batch size, spike happens once per epoch

<img width="428" alt="screenshot 2018-08-02 14 26 11" src="https://user-images.githubusercontent.com/23068/43612069-041def0e-9660-11e8-9a23-829c19f02b9b.png">

https://github.com/diux-dev/cluster/blob/master/pytorch/recipes-yaroslav-va.ipynb

**Comment by bearpelican** [comment]
Spikes seem to have been resolved in tensorflow. Closing until it reappears
